File: src/main/java/nl/shadowlink/tools/shadowlib/ipl/IPL_Item.java
package nl.shadowlink.tools.shadowlib.ipl;

import com.nikhaldimann.inieditor.IniEditor;
import nl.shadowlink.tools.io.ReadFunctions;
import nl.shadowlink.tools.shadowmapper.utils.hashing.HashTable;

/**
 * @author Shadow-Link
 */
public abstract class IPL_Item {

    public abstract void read(String line);

    public abstract void read(ReadFunctions rf);

    public abstract void read(ReadFunctions rf, HashTable hashTable);

}

File: src/main/java/nl/shadowlink/tools/shadowlib/ipl/IplItem.kt
package nl.shadowlink.tools.shadowlib.ipl

import nl.shadowlink.tools.io.ReadFunctions
import nl.shadowlink.tools.shadowmapper.utils.hashing.HashTable

/**
 * @author Shadow-Link
 */
abstract class IplItem {
    abstract fun read(line: String?)
    abstract fun read(rf: ReadFunctions?)
    abstract fun read(rf: ReadFunctions?, hashTable: HashTable?)
}